server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -110,36 +119,70 @@ private void verifyMotifSearchRequest(final MotifSearchRequest motifSearchReques
 
     private MotifSearchResult searchRegionMotifs(final MotifSearchRequest request) {
         final Chromosome chromosome = loadChrById(request.getReferenceId(), request.getChromosomeId());
+        final byte[] sequence;
+        try {
+            sequence= referenceManager.getSequenceByteArray(request.getStartPosition(),
+                            request.getEndPosition(), request.getReferenceId(), chromosome.getName());
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to read reference data by ID: " + request.getReferenceId(), e);
+        }
+        final List<Motif> searchResult =
+                MotifSearcher.search(sequence, request.getMotif(), request.getStrand(),
+                        chromosome.getName(), request.getStartPosition());
         return MotifSearchResult.builder()
-                .result(
-                    fillMotifList(
-                        chromosome, request.getStartPosition(),
-                        request.getEndPosition(), 0,
-                            request.getStrand()))
+                .result(searchResult)
                 .chromosomeId(request.getChromosomeId())
-                .pageSize(request.getPageSize())
-                .position(request.getEndPosition())
+                .pageSize(searchResult.size())
+                .position(searchResult.get(searchResult.size()-1).getStart())
```

Comment:
please use ` ` between operands and operators to be complaint with code conventions -> `size() - 1`